csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -170,6 +183,24 @@ public void Format(IMessage message, TextWriter writer)
         }
 
         /// <summary>
+        /// Formats the specified field of the message as JSON.
+        /// </summary>
+        /// <param name="message">The message to format.</param>
+        /// <param name="field">The field of the message to format.</param>
+        /// <param name="writer">The TextWriter to write the formatted message to.</param>
+        /// <returns>The formatted field.</returns>
+        public void Format(IMessage message, FieldDescriptor field, TextWriter writer)
+        {
+            ProtoPreconditions.CheckNotNull(message, nameof(message));
+            ProtoPreconditions.CheckNotNull(field, nameof(field));
+            ProtoPreconditions.CheckNotNull(writer, nameof(writer));
```

Comment:
Validate that the field actually belongs to the message type? (i.e. `field.ContainingType == message.Descriptor`)